### PR TITLE
Make the docs compatible with RTD and update README

### LINKED ISSUE #252: Separate the README and the PyPI documentation
It would be better if we could just upload a subset of the README to PyPI. It looks too crowded.

See the https://www.npmjs.com/package/hazelcast-client and https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/DOCUMENTATION.md for example.

### PR DESCRIPTION
**Pull request by mdumandag**
It makes more sense to host API documentation and the long README
we have in Read the Docs so that they become more visible and
structured. Therefore, the long README is partitioned into chapters,
a small introduction(docs/index.rst) is created, and API documentation
is moved into a separate chapter. Also, API documentation of the
HazelcastClient is constructed as a top-level chapter since it is
generally the entry point for other documentation elements.

Finally, the README we have in Github is made shorter and
more descriptive. Also, added some badges to both README and
docs/index.rst to make users easily navigate between PyPI, Read the Docs,
and Github. The links to Read the Docs site (hazelcast.readthedocs.io)
won't work right now, but I will set up this once this is merged.

Closes #252

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,3784 +1,221 @@
...
-    2020-10-16 13:31:35,605 - hazelcast.lifecycle - INFO - HazelcastClient 4.0.0 is STARTED
-    2020-10-16 13:31:35,605 - hazelcast.connection - INFO - Trying to connect to Address(host=127.0.0.1, port=5701)
-    2020-10-16 13:31:35,622 - hazelcast.lifecycle - INFO - HazelcastClient 4.0.0 is CONNECTED
-    2020-10-16 13:31:35,622 - hazelcast.connection - INFO - Authenticated with server Address(host=172.17.0.2, port=5701):7682c357-3bec-4841-b330-6f9ae0c08253, server version: 4.0, local address: Address(host=127.0.0.1, port=56752)
-    2020-10-16 13:31:35,623 - hazelcast.cluster - INFO -
+    # Try to get the value associated with the given key from the cluster
+    # and attach a callback to be executed once the response for the
+    # get request is received. Note that, the set request above was
+    # blocking since it calls ".result()" on the returned Future, whereas
+    # the get request below is non-blocking.
```
Comment (puzpuzpuz) [inline README.rst:91]: nit: probably, a redundant line here.

File: README.rst [hunk 2]
```diff
@@ -1,3784 +1,221 @@
...
-    }
 
-    2020-10-16 13:31:35,624 - hazelcast.client - INFO - Client started
-    2020-10-16 13:31:35,624 - hazelcast.lifecycle - INFO - HazelcastClient 4.0.0 is SHUTTING_DOWN
-    2020-10-16 13:31:35,624 - hazelcast.connection - INFO - Removed connection to Address(host=127.0.0.1, port=5701):7682c357-3bec-4841-b330-6f9ae0c08253, connection: Connection(id=0, live=False, remote_address=Address(host=172.17.0.2, port=5701))
-    2020-10-16 13:31:35,624 - hazelcast.lifecycle - INFO - HazelcastClient 4.0.0 is DISCONNECTED
-    2020-10-16 13:31:35,634 - hazelcast.lifecycle - INFO - HazelcastClient 4.0.0 is SHUTDOWN
+    def callback(future):
+        # Outputs "value"
+        print(future.result())
```
Comment (puzpuzpuz) [inline README.rst:96]: nit: probably, a redundant line here.

File: README.rst [hunk 3]
```diff
@@ -1,3784 +1,221 @@
...
+Issue Reports
+~~~~~~~~~~~~~
 
-On the client side, you have to provide ``ssl_cafile``, ``ssl_certfile``
-and ``ssl_keyfile`` on top of the other TLS/SSL configurations. See the
-`TLS/SSL for Hazelcast Python
-Clients <#812-tlsssl-for-hazelcast-python-clients>`__ for the details of
-these options.
+For issue reports, please share the following information with us to
+quickly resolve the problems.
```
Comment (puzpuzpuz) [inline README.rst:184]:
```suggestion
quickly resolve the problems:
```

File: README.rst [hunk 4]
```diff
@@ -1,3784 +1,221 @@
...
 
-9. Development and Testing
-==========================
+-  Hazelcast IMDG and the client version that you use
+-  General information about the environment and the architecture you
+   use like Python version, cluster size, number of clients, Java
+   version, JVM parameters, operating system etc.
+-  Logs and stack traces, if any.
```
Comment (puzpuzpuz) [inline README.rst:190]:
```suggestion
-  Logs and stack traces, if any
```

File: README.rst [hunk 5]
```diff
@@ -1,3784 +1,221 @@
...
+-  Detailed description of the steps to reproduce the issue.
```
Comment (puzpuzpuz) [inline README.rst:191]:
```suggestion
-  Detailed description of the steps to reproduce the issue
```

File: README.rst [hunk 6]
```diff
@@ -1,3784 +1,221 @@
...
+-  Distributed, CRDT based counter, called **PNCounter**
+-  Distributed concurrency primitives from CP Subsystem such as
+   **FencedLock**, **Semaphore**, **AtomicLong**
+-  Integration with `Hazelcast Cloud <https://cloud.hazelcast.com/>`__
+-  Support for serverless and traditional web service architectures with
+   **Unisocket** and **Smart** operation modes
+-  Ability to listen client lifecycle, cluster state and distributed
+   data structure events
+-  and `many
+   more <https://hazelcast.org/imdg/clients-languages/python/#client-features>`__.
```
Comment (puzpuzpuz) [inline README.rst:158]:
```suggestion
   more <https://hazelcast.org/imdg/clients-languages/python/#client-features>`__
```

File: README.rst [hunk 7]
```diff
@@ -1,3784 +1,221 @@
...
-If you want to help with bug fixes, develop new features or tweak the
-implementation to your application’s needs, you can follow the steps in
-this section.
+Pull Requests
+~~~~~~~~~~~~~
 
-9.1. Building and Using Client From Sources
--------------------------------------------
+Contributions are submitted, reviewed and accepted using the pull
+requests on GitHub. For an enhancement or larger feature, create a
```
Comment (puzpuzpuz) [inline README.rst:197]:
```suggestion
requests on GitHub. For an enhancement or larger feature, please create a
```

**Review by puzpuzpuz — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
Left some comments, all nits.

File: README.rst (hunk 1 above)
Comment (mdumandag) [reply README.rst:91]: I don't like this extra space too but PEP8 recommends it 
https://www.python.org/dev/peps/pep-0008/#blank-lines 
`Surround top-level function and class definitions with two blank lines.`

Comment (puzpuzpuz) [reply README.rst:91]: Didn't know that. Let's leave it as is then.

**Comment by mdumandag** [comment]
Thanks for the review Andrey

**Pull request merged**